streams/src/main/java/org/apache/kafka/streams/processor/internals/SinkNode.java

```diff
@@ -58,8 +60,21 @@ public void addChild(final ProcessorNode<Void, Void, ?, ?> child) {
     public void init(final InternalProcessorContext<Void, Void> context) {
         super.init(context);
         this.context = context;
-        keySerializer = prepareKeySerializer(keySerializer, context, this.name());
-        valSerializer = prepareValueSerializer(valSerializer, context, this.name());
+        try {
+            keySerializer = prepareKeySerializer(keySerializer, context, this.name());
+        } catch (final ConfigException e) {
+            throw new ConfigException(String.format("Failed to initialize key serdes for sink node %s", name()));
+        } catch (final StreamsException e) {
```

Comment:
Why are we catching `StreamsException`?  Seems the only exception that might bubble up her is a `ConfigException`?

Reply:
prepareSerializer calls context.keySerde which calls config.defaultKeySerde(). The latter can throw both exceptions

Reply:
Wondering if we should merge both blocks and just catch `RuntimeException`? Seems to be duplicate code? And just pass in whatever exception we catch and rethrow as `new StreamsException(..., e);`

Reply:
@mjsax the idea was to preserve the exception that was thrown, as I think StreamsException and ConfigException don't have the same nature. AFAIK StreamsException captures problems related to the state of the task. If you believe it's ok to throw StreamsException here, I can do the change

Reply:
Can't say what it's original purpose was but StreamsException has definitely morphed into a catch-all for exceptions throughout Streams. It's definitely not exclusive to the state of a task though (that would be ProcessorStateException). The nice thing about StreamsException is you can add other useful metadata such as the taskId where the error originated, so I always prefer to just throw the StreamsException. We also know for a fact that StreamsException will be caught and handled properly as it gets bubbled up. So I'd go for merging this into a single `catch RuntimeException` block, then wrap it in a StreamsException. 

And don't forget to add the task id too! 😄  (you can get it from the processor context)

Reply:
Why `RuntimeException`? Can't we catch `ConfigException` and `StreamsException` in one catch clause and then throw a `StreamsException`. This seems safer to me, because otherwise with future code changes that might throw exceptions like `IllegalStateException` we would wrap unexpected exception due to mistakes in Streams in `StreamsException`.

```java
catch (final ConfigException | final StreamsException ex) {
    // handle the exception
}
```

Reply:
Good point -- I agree with Bruno, the catch block should be for just ConfigException or StreamsException